src/Turing.jl

```diff
@@ -24,30 +24,29 @@ using Markdown
 using Libtask
 using MacroTools
 
-function __init__()
-    @require CmdStan="593b3428-ca2f-500c-ae53-031589ec8ddd" @eval begin
-        using CmdStan
-        import CmdStan: Adapt, Hmc
-    end
+@init @require CmdStan="593b3428-ca2f-500c-ae53-031589ec8ddd" @eval begin
+    using CmdStan
+    import CmdStan: Adapt, Hmc
+end
 
-    @require DynamicHMC="bbc10e6e-7c05-544b-b16e-64fede858acb" @eval begin
-        using DynamicHMC, LogDensityProblems
-        using LogDensityProblems: AbstractLogDensityProblem, ValueGradient
+@init @require DynamicHMC="bbc10e6e-7c05-544b-b16e-64fede858acb" @eval begin
+    using .DynamicHMC: NUTS_init_tune_mcmc
```

Comment:
Is `.` in `.DynamicHMC` necessary here? Seems you didn't use `.` when loading `LogDensityProblems`.

Reply:
The . is from the Requires.jl README.

> In the `@require` block, or any included files, you can use or import the package, but note that you must use the syntax `using .Gadfly` or `import .Gadfly`, rather than the usual syntax. Otherwise you will get a warning about `Gadfly` not being in dependencies.

The problem with `LogDensityProblems` is that  `Requires` only "loads" `DynamicHMC` by default using the line: `const DynamicHMC = Base.require(Base.PkgId(Base.UUID("bbc10e6e-7c05-544b-b16e-64fede858acb"), "DynamicHMC"))`, which is generated by the `@require` block, bringing only `DynamicHMC` to the namespace. `DynamicHMC` has `LogDensityProblems` in its deps but it only does `using LogDensityProblems: ...`, I cannot bring in `LogDensityProblems` using `DynamicHMC.LogDensityProblems`. 

The only other 2 ways are: 1) to do `using LogDensityProblems` which needs `LogDensityProblems` to be in the `REQUIRE` file, or 2) to use the Requires.jl approach above. The Requires.jl approach seems to work even when the package is not in the `Project.toml` of the environment and just loads the package without complaining about deps. I am not exactly sure how it works tbh, but it seems kind of hackish, since it bypasses regular dependency checks. As you can see, I went with the first approach, but the second is also a candidate, hoping it won't break things. I can ask around Slack to see if it is safe to use or not.

Reply:
Perhaps split the block into the following? 

```julia

@init @require DynamicHMC="bbc10e6e-7c05-544b-b16e-64fede858acb" @eval begin
    using .DynamicHMC: NUTS_init_tune_mcmc
end

@init @require LogDensityProblems="some unique id" @eval begin
    using .LogDensityProblems: LogDensityProblems, AbstractLogDensityProblem, ValueGradient

    struct FunctionLogDensity{F} <: AbstractLogDensityProblem
      dimension::Int
      f::F
    end

    LogDensityProblems.dimension(ℓ::FunctionLogDensity) = ℓ.dimension

    LogDensityProblems.logdensity(::Type{ValueGradient}, ℓ::FunctionLogDensity, x) = ℓ.f(x)::ValueGradient
end
```

Reply:
This is ok but the only problem with this is that the examples using both `LogDensityProblems` and `DynamicHMC` would need to start with:
```julia
using LogDensityProblems, DynamicHMC, Turing
```

Reply:
`using LogDensityProblems, DynamicHMC, Turing` should be fine.

BTW, does `using DynamicHMC` load `LogDensityProblems` into Julia?